### PR TITLE
Add dgram.Socket.send overloads

### PR DESCRIPTION
**Pull request by stijnvn**
Add overloads for optional parameters and first parameter with type ```string```.

### CODE REVIEW

File: 4/node.d.ts [hunk 1]
```diff
@@ -1363,6 +1363,13 @@ declare module "dgram" {
 
     interface Socket extends events.EventEmitter {
         send(buf: Buffer, offset: number, length: number, port: number, address: string, callback?: (error: Error, bytes: number) => void): void;
+        send(buf: Buffer, offset: number, port: number, address: string, callback?: (error: Error, bytes: number) => void): void;
+        send(buf: Buffer, length: number, port: number, address: string, callback?: (error: Error, bytes: number) => void): void;
+        send(buf: Buffer, port: number, address: string, callback?: (error: Error, bytes: number) => void): void;
+        send(str: string, offset: number, length: number, port: number, address: string, callback?: (error: Error, bytes: number) => void): void;
+        send(str: string, offset: number, port: number, address: string, callback?: (error: Error, bytes: number) => void): void;
+        send(str: string, length: number, port: number, address: string, callback?: (error: Error, bytes: number) => void): void;
+        send(str: string, port: number, address: string, callback?: (error: Error, bytes: number) => void): void;
```
Comment (blakeembrey) [inline 4/node.d.ts]: LGTM. Would it be better to use a union? Reference: https://nodejs.org/api/dgram.html#dgram_socket_send_msg_offset_length_port_address_callback. E.g.

```ts
send(buf: string | Buffer | Array<any>, port: number, address: string, callback?: (error: Error, bytes: number) => void): void;
send(buf: string | Buffer, offset: number, length: number, port: number, address: string, callback?: (error: Error, bytes: number) => void): void;
```

(Not sure in the docs what array type it's mean to be.)

Comment (stijnvn) [inline 4/node.d.ts]: I have changed the signatures to use a union type.

File: 4/node.d.ts [hunk 2]
```diff
@@ -1362,7 +1362,10 @@ declare module "dgram" {
     export function createSocket(type: string, callback?: (msg: Buffer, rinfo: RemoteInfo) => void): Socket;
 
     interface Socket extends events.EventEmitter {
-        send(buf: Buffer, offset: number, length: number, port: number, address: string, callback?: (error: Error, bytes: number) => void): void;
+        send(msg: Buffer | string | Array<any>, offset: number, length: number, port: number, address: string, callback?: (error: Error, bytes: number) => void): void;
+        send(msg: Buffer | string | Array<any>, offset: number, port: number, address: string, callback?: (error: Error, bytes: number) => void): void;
```
Comment (louy) [inline 4/node.d.ts]: In the docs it says that both `offset` and `length` are optional but should either be put together or skipped together. i.e. this overload and the next one aren't needed.

Comment (louy) [inline 4/node.d.ts]: also it seems like msg can either be a `Buffer`, a `String` or `Array<Buffer | String>`.

**Comment by blakeembrey** [comment]
LGTM /cc @louy

**Pull request merged**